scipy/stats/_continuous_distns.py

```diff
@@ -2976,6 +2976,46 @@ def _entropy(self):
         # https://en.wikipedia.org/wiki/Gumbel_distribution
         return _EULER + 1.
 
+    def fit(self, data, *args, **kwds):
+        data, floc, fscale = _check_fit_input_parameters(self, data,
+                                                         args, kwds)
+
+        # if user has provided `floc` or `fscale`, fall back on super fit
+        # method. This scenario is not suitable for solving a system of
+        # equations
+        if floc is not None or fscale is not None:
+            return super(gumbel_r_gen, self).fit(data, *args, **kwds)
+
+        # rv_continuous provided guesses
+        loc, scale = self._fitstart(data)
+        # account for user provided guesses
+        loc = kwds.pop('loc', loc)
+        scale = kwds.pop('scale', scale)
+
+        # By the method of maximum likelihood, the estimators `a` and `b` of
+        # the location and scale are the roots of the simultaneous
+        # equations described in `func`. Source: Statistical Distributions,
+        # 3rd Edition. Evans, Hastings, and Peacock (2000), Page 101
+
+        def func(vals, data):
+            a, b = vals
+            n = len(data)
+            exp_x_b = np.exp(-data / b)
+            sum_exp_x_b = np.sum(exp_x_b)
+            x1 = data.mean() - (np.dot(data, exp_x_b) / sum_exp_x_b) - b
+            x2 = - b * np.log(sum_exp_x_b / n) - a
+            return x1, x2
+
+        # some data with scale approaching zero cause invalid operations
+        with warnings.catch_warnings():
+            warnings.filterwarnings(action='ignore',
+                                    category=RuntimeWarning)
```

Comment:
There's a test error in `test_warning_calls_filters` which I think is related to this line.

See https://github.com/scipy/scipy/blob/master/scipy/_lib/tests/test_warnings.py#L84

```
[gw1] win32 -- Python 3.6.8 C:\hostedtoolcache\windows\Python\3.6.8\x64\python.exe
C:\hostedtoolcache\windows\Python\3.6.8\x64\lib\site-packages\scipy\_lib\tests\test_warnings.py:76: in warning_calls
    finder.visit(tree)
        bad_filters = ['optimize\\optimize.py:947', 'sparse\\__init__.py:245']
        bad_stacklevels = ['__init__.py:135', 'cluster\\hierarchy.py:3528', 'cluster\\vq.py:135', 'cluster\\vq.py:575', 'cluster\\vq.py:737', 'constants\\codata.py:1568', ...]
        base       = WindowsPath('C:/hostedtoolcache/windows/Python/3.6.8/x64/lib/site-packages/scipy')
        file       = <_io.TextIOWrapper name='C:\\hostedtoolcache\\windows\\Python\\3.6.8\\x64\\lib\\site-packages\\scipy\\stats\\_continuous_distns.py' mode='r' encoding='utf-8'>
        finder     = <scipy._lib.tests.test_warnings.FindFuncs object at 0x0000020F0773B550>
        path       = WindowsPath('C:/hostedtoolcache/windows/Python/3.6.8/x64/lib/site-packages/scipy/stats/_continuous_distns.py')
        tree       = <_ast.Module object at 0x0000020F07734FD0>
C:\hostedtoolcache\windows\Python\3.6.8\x64\lib\ast.py:253: in visit
    return visitor(node)
        method     = 'visit_Module'
        node       = <_ast.Module object at 0x0000020F07734FD0>
        self       = <scipy._lib.tests.test_warnings.FindFuncs object at 0x0000020F0773B550>
        visitor    = <bound method NodeVisitor.generic_visit of <scipy._lib.tests.test_warnings.FindFuncs object at 0x0000020F0773B550>>
C:\hostedtoolcache\windows\Python\3.6.8\x64\lib\ast.py:261: in generic_visit
```